src/Entity/Auth/User.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace App\Entity\Auth;
+
+use Cocur\Slugify\SlugifyInterface;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Ramsey\Uuid\Uuid;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * @ORM\Entity(repositoryClass="App\Repository\Auth\UserRepository")
+ */
+class User implements UserInterface
+{
+    /**
+     * @ORM\Id()
+     * @ORM\GeneratedValue(strategy="CUSTOM")
+     * @ORM\Column(type="uuid", unique=true)
+     * @ORM\CustomIdGenerator(class="Ramsey\Uuid\Doctrine\UuidGenerator")
+     *
+     * @var Uuid
+     */
+    protected $id;
+
+    /**
+     * @ORM\OneToMany(targetEntity="App\Entity\Auth\UserCharacter", mappedBy="user", orphanRemoval=true)
+     */
+    protected $characters;
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     */
+    protected $username;
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     */
+    protected $slug;
+
+    /**
+     * @var SlugifyInterface
+     */
+    protected $slugify;
+
+    /**
+     * User constructor.
+     *
+     * @param SlugifyInterface $slugify
+     */
+    public function __construct(SlugifyInterface $slugify)
+    {
+        $this->slugify = $slugify;
```

Comment:
I'm not sure if i would treat my entities as services...

I think it would be more clean have a manager / command / whatever in the middle so 

```
UserManager::create($username, ....)

$slug = UserManager::$slugify->slugify($username);
$user->setSlug($slug);

// persist
// flush
```

Reply:
You're totally right on that !
I don't like this `Manager` name, I think I'm gonna use `UserFactory` but not sure if `Factory` is really related

Reply:
Is really up to you, i like to have a manager for each entity - sort of a middle-ware between my app and entity (entity manager) so i can trigger events, logs or any other domain logic in between (this also helps testing mutations).

If you only have a single creation of user and this don't really do anything else than receiving some arguments, not sure how "useful" would be a factory be.